datafusion/expr/src/udaf.rs

```diff
@@ -389,6 +410,13 @@ pub trait AggregateUDFImpl: Debug + Send + Sync {
         &[]
     }
 
+    fn create_sliding_accumulator(
```

Comment:
Could we please add some documentation here explaining what this is?

Basically I think this is to allow returning a different [`Accumulator` ](https://docs.rs/datafusion/latest/datafusion/logical_expr/trait.Accumulator.html)instance that is optimized for sliding windows (e.g. incrementally computing output via https://docs.rs/datafusion/latest/datafusion/logical_expr/trait.Accumulator.html#method.retract_batch